java/src/main/java/com/spotify/featran/java/SerializableFunction.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.java;
+
+import java.io.Serializable;
+
+public interface SerializableFunction<InputT, OutputT> extends Serializable {
```

Comment:
You can use lambda syntax at call site i.e. `.required(r -> r.getValue)` but `java.util.Function` is not serializable so this is a hack as argument type for `required`. It's actually copied as is from Beam.
Yes Java 8 is full of sadness. Everything (optional, stream, future) feels half done.